### ISSUE TITLE
custom credential verification for auth service google failed

### ISSUE THREAD

**Issue by techyrajeev** [ParsePlatform/parse-server#2766]
I have enable oauth support on parse-server.
I have set this in configuration of parse-server.
```
  oauth: {
        google: true
   }
```

I am getting error like - 
`custom credential verification for auth service google failed errorCode: 251
`

Flow is like this - 

I have a separate server where I get auth token for google. 
From there I form authData and link it to parse via rest api call.

Similar to this

```
curl -X PUT \
  -H "X-Parse-Application-Id: ${APPLICATION_ID}" \
  -H "X-Parse-REST-API-Key: ${REST_API_KEY}" \
  -H "X-Parse-Session-Token: r:samplei3l83eerhnln0ecxgy5" \
  -H "Content-Type: application/json" \
  -d '{
        "authData": {
          "google": {
            "id": "123456789",
            "access_token": "SaMpLeAAibS7Q55FSzcERWIEmzn6rosftAr7pmDME10008bWgyZAmv7mziwfacNOhWkgxDaBf8a2a2FCc9Hbk9wAsqLYZBLR995wxBvSGNoTrEaL",
            "expiration_date": "2012-02-28T23:49:36.353Z"
          }
        }
      }' \
  https://api.parse.com/1/users/uMz0YZeAqc
```

Server setup:

```
var api = new ParseServer({
  databaseURI: databaseUri || 'mongodb://localhost:27017/dev',
  cloud: process.env.CLOUD_CODE_MAIN || __dirname + '/cloud/main.js',
  appId: process.env.APP_ID || 'myAppId',
  masterKey: process.env.MASTER_KEY || '', //Add your master key here. Keep it secret!
  serverURL: process.env.SERVER_URL',  // Don't forget to change to https if needed
  liveQuery: {
    classNames: ["Posts", "Comments"] // List of classes to support for query subscriptions
  }
});
```
Steps to reproduce:

1. Deploy Parse-Server to Heroku.
2. Use google server side authentication verifaction to validate  authToken required ( Not on parse server)
3. If user authentication successful get authData object
4. link it with parse user via sending rest call to parse 

send curl request

```
curl -X PUT \
  -H "X-Parse-Application-Id: ${APPLICATION_ID}" \
  -H "X-Parse-REST-API-Key: ${REST_API_KEY}" \
  -H "X-Parse-Session-Token: r:samplei3l83eerhnln0ecxgy5" \
  -H "Content-Type: application/json" \
  -d '{
        "authData": {
          "google": {
            "id": "123456789",
            "access_token": "SaMpLeAAibS7Q55FSzcERWIEmzn6rosftAr7pmDME10008bWgyZAmv7mziwfacNOhWkgxDaBf8a2a2FCc9Hbk9wAsqLYZBLR995wxBvSGNoTrEaL",
            "expiration_date": "2012-02-28T23:49:36.353Z"
          }
        }
      }' \
  https://api.parse.com/1/users/uMz0YZeAqc
```

Expected Results

User object needs to be created on parse with authData provided by google.
Actual Outcome
Error like - 
`custom credential verification for auth service google failed errorCode: 251
`
Environment Setup

Server

express: "^4.14.0",
parse: "^1.9.1",
parse-server: "^2.2.18"

Operating System: Mac Os
Hardware: MacBook Pro
Localhost or remote server? (AWS, Heroku, Azure, Digital Ocean, etc): Heroku
Database

MongoDB version: 3.0.12
Storage engine: AWS us-east-1
Localhost or remote server? (AWS, mLab, ObjectRocket, Digital Ocean, etc): mLab

**Comment by flovilmart** [comment]
What was the content of your auth data? Which keys did you set?

**Comment by techyrajeev** [comment]
I had set authData to
{ "id" : ID'
access_token: token from Google
}

**Comment by hramos** [comment]
Hey, thanks for reporting this issue.

Can you please add all the info specified in the [template](https://github.com/ParsePlatform/parse-server/blob/master/.github/ISSUE_TEMPLATE.md)? This is necessary for people to be able to understand and reproduce the issue being reported.

**Comment by techyrajeev** [comment]
Update the issue with all necessary details.

**Comment by steven-supersolid** [comment]
It looks like your curl is still hitting parse.com. Parse does not support custom auth unless you write it yourself, only Facebook auth.

Also where are you passing?
```
oauth: {
        google: true
   }
```
I do not see this in your server config. Your server should look more like the following if this were the case:
```
var api = new ParseServer({
  databaseURI: databaseUri || 'mongodb://localhost:27017/dev',
  cloud: process.env.CLOUD_CODE_MAIN || __dirname + '/cloud/main.js',
  appId: process.env.APP_ID || 'myAppId',
  masterKey: process.env.MASTER_KEY || '', //Add your master key here. Keep it secret!
  serverURL: process.env.SERVER_URL',  // Don't forget to change to https if needed
  liveQuery: {
    classNames: ["Posts", "Comments"] // List of classes to support for query subscriptions
  },
  oauth: {
        google: true
  }
});
```
If it is, I would try removing the oauth option as it should not be required for google as this a default auth provider and I think has no options.

Please also post the exact version of parse-server you are using.

**Comment by flovilmart** [comment]
THis should be resolved now with 2.2.23. Please reopen if it isn't.
